elfpy/data/postgres.py

```diff
@@ -326,16 +326,16 @@ def get_transactions(session: Session, start_block: int | None = None, end_block
 
     # Support for negative indices
     if (start_block is not None) and (start_block < 0):
-        start_block = _get_latest_block_number_transactions(session) + start_block + 1
+        start_block = get_latest_block_number_from_table(Transaction.__tablename__, session) + start_block + 1
     if (end_block is not None) and (end_block < 0):
-        end_block = _get_latest_block_number_transactions(session) + end_block + 1
+        end_block = get_latest_block_number_from_table(Transaction.__tablename__, session) + end_block + 1
 
     if start_block is not None:
         query = query.filter(Transaction.blockNumber >= start_block)
     if end_block is not None:
         query = query.filter(Transaction.blockNumber < end_block)
 
-    return pd.read_sql(query.statement, con=session.connection())
+    return pd.read_sql(query.statement, con=session.connection()).set_index("blockNumber")
```

Comment:
I think this is fine, but just a reminder that in the transactions, blockNumber isn't unique. Pandas is able to handle non-unique indices in most functions, but I do think there are limits to this.

Reply:
yes once/if we have more than 1 transaction per block, this would need to be removed, maybe to a multi-index of blockNumber and transactionIndex. we currently don't support that however.